src/core/public/chrome/ui/header/home_icon.test.tsx

```diff
@@ -0,0 +1,255 @@
+        mark: { defaultUrl: '/defaultModeMark', darkModeUrl: '/darkModeMark' },
+        applicationTitle: 'custom title',
+        assetFolderUrl: 'base/ui/default_branding',
+        useExpandedMenu: false,
+      };
+      const component = mountWithIntl(<HomeIcon {...branding} />);
+      const icon = component.find('EuiIcon');
+      expect(icon.prop('type')).toEqual(branding.mark.darkModeUrl);
+      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} home`);
+      expect(component).toMatchSnapshot();
+    });
+  });
+
+  describe('in light mode ', () => {
+    it('uses home icon if no branding provided', () => {
+      const branding = {};
+      const component = mountWithIntl(<HomeIcon {...branding} />);
+      const icon = component.find('EuiIcon');
+      expect(icon.prop('type')).toEqual('home');
+      expect(icon.prop('size')).toEqual(`m`);
+      expect(icon.prop('title')).toEqual(`opensearch dashboards home`);
+      expect(component).toMatchSnapshot();
+    });
+
+    it('uses home icon if no mark provided', () => {
+      const branding = {
+        darkMode: false,
+        logo: {},
+        mark: {},
+        applicationTitle: 'custom title',
+        assetFolderUrl: 'base/ui/default_branding',
+      };
+      const component = mountWithIntl(<HomeIcon {...branding} />);
+      const icon = component.find('EuiIcon');
+      expect(icon.prop('type')).toEqual('home');
+      expect(icon.prop('size')).toEqual(`m`);
+      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} home`);
+      expect(component).toMatchSnapshot();
+    });
+
+    it('uses home icon if custom logo provided without mark', () => {
+      const branding = {
+        darkMode: false,
+        logo: { defaultUrl: '/defaultModeLogo' },
+        mark: {},
+        applicationTitle: 'custom title',
+        assetFolderUrl: 'base/ui/default_branding',
+      };
+      const component = mountWithIntl(<HomeIcon {...branding} />);
+      const icon = component.find('EuiIcon');
+      expect(icon.prop('type')).toEqual('home');
+      expect(icon.prop('size')).toEqual(`m`);
+      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} home`);
+      expect(component).toMatchSnapshot();
+    });
+
+    it('uses custom mark default mode URL', () => {
+      const branding = {
+        darkMode: false,
+        logo: {},
+        mark: { defaultUrl: '/defaultModeMark' },
+        applicationTitle: 'custom title',
+        assetFolderUrl: 'base/ui/default_branding',
+      };
+      const component = mountWithIntl(<HomeIcon {...branding} />);
+      const icon = component.find('EuiIcon');
+      expect(icon.prop('type')).toEqual(branding.mark.defaultUrl);
+      expect(icon.prop('size')).toEqual(`l`);
+      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} home`);
+      expect(component).toMatchSnapshot();
+    });
+  });
+
+  describe('in dark mode ', () => {
+    it('uses home icon if no mark provided', () => {
```

Comment:
nit: Wondering if there is a nicer way to test out so many combinations? You can probably try using `it.each` to iterate though the different combinations so that refactoring in future is not a nightmare.

Reply:
Yeah, agreed - I didn't update the structure of these tests, but wanted to make sure all the conditions were properly covered. I added a todo that links to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/895#issuecomment-1164995007